### PR TITLE
feat(medusa, core-workflows, product): slightly improve create cart workflow

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
Improve create cart workflow, passing the entity instead of just the id when we have the entity at hand will prevent the create step to do the entire work a second time from within the cart service. The workflow have been design to split the creation into multiple steps, calling the cart service create method will also do the same step as we are currently only providing ids. On simple cart creation involving the region I got a 15% better response time from 108ms to 92ms after 600 requests for 45s

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

File: packages/medusa/src/types/cart.ts [hunk 1]
```diff
@@ -60,6 +60,7 @@ export type CartCreateProps = {
   shipping_address?: Partial<AddressPayload>
   gift_cards?: GiftCard[]
   discounts?: Discount[]
+  customer?: any
```
Comment (olivermrbl) [inline packages/medusa/src/types/cart.ts]: question: Do we need a type?

File: packages/core-flows/src/handlers/region/find-region.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { WorkflowArguments } from "@medusajs/workflows-sdk"
 
 type RegionDTO = {
   region_id?: string
+  region?: any
```
Comment (olivermrbl) [inline packages/core-flows/src/handlers/region/find-region.ts]: question: Add type?

File: packages/core-flows/src/handlers/customer/find-or-create-customer.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { validateEmail } from "@medusajs/utils"
 import { WorkflowArguments } from "@medusajs/workflows-sdk"
 
 type CustomerDTO = {
+  customer?: any
```
Comment (olivermrbl) [inline packages/core-flows/src/handlers/customer/find-or-create-customer.ts]: question: Add type? We could change the DTO name too to better reflect its purpose

**Review by olivermrbl — APPROVED** [review]

File: packages/core-flows/src/handlers/cart/create-cart.ts [hunk 4]
```diff
@@ -14,14 +14,18 @@ type HandlerInputData = {
     sales_channel_id?: string
   }
   addresses: {
+    shipping_address?: AddressDTO
     shipping_address_id: string
+    billing_address?: AddressDTO
     billing_address_id: string
   }
   customer: {
+    customer?: any
```
Comment (olivermrbl) [inline packages/core-flows/src/handlers/cart/create-cart.ts]: question: what about customer and region types here?

Comment (adrien2p) [reply packages/core-flows/src/handlers/cart/create-cart.ts]: done + finally fully fixed the product module integration tests 🚀

**Pull request merged**